### PR TITLE
[ie/TVer] Fix extractor

### PR DESCRIPTION
**Pull request by vvto33**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

1. Support for series URL
Added support for downloading from show-specific URLs instead of per-episode URLs, similar to #3075.
- https://tver.jp/series/srkq2shp9d (ONE PIECE)
- https://tver.jp/series/srtxft431v (Detective Conan / Case Closed)

2. Add thumbnail URL support
This addresses situations where the extracted thumbnail was not representative of the actual video content, and instead showed a generic default image from the broadcaster. With this change, yt-dlp will now fetch the appropriate thumbnail for the specific video.
- https://tver.jp/episodes/epfauhb2w7
  -  Old: https://cf-images.ap-northeast-1.prod.boltdns.net/v1/static/4394098883001/2f80cf99-1cf9-4d74-923c-3a930af7f892/01d4088e-555a-4ca7-a5b8-ec6dfc80d46a/1920x1080/match/image.jpg
  -  New: https://statics.tver.jp/images/content/thumbnail/episode/xlarge/epfauhb2w7.jpg?v=32
- https://tver.jp/episodes/epsvoup1sf
  -  Old: https://cf-images.ap-northeast-1.prod.boltdns.net/v1/static/4394098883001/b43ffdbb-a4d9-4b78-a7c9-f51c7b29132b/278b0362-d9b1-4a86-bded-24065426345f/1920x1080/match/image.jpg
  -  New: https://statics.tver.jp/images/content/thumbnail/episode/xlarge/epsvoup1sf.jpg?v=12

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by pzhlkj6612** [comment]
Merging this PR will close #3727.

**Comment by pzhlkj6612** [comment]
Hi, there! I'm going to add new code by creating a PR to @vvto33's repository. Otherwise, those ruff checks will fail after merging to the master.

**Comment by bashonly** [comment]
@pzhlkj6612 do you want me to merge master into the branch before you do that?

**Comment by pzhlkj6612** [comment]
> @pzhlkj6612 do you want me to merge master into the branch before you do that?

It's OK!

**Comment by bashonly** [comment]
So I should?

**Comment by pzhlkj6612** [comment]
@bashonly 

> So I should?

Sorry I hit Ctrl-Enter too quickly. Please merge master into the branch. I'll add new code later.

**Comment by bashonly** [comment]
@pzhlkj6612 Done. Thanks!

**Comment by pzhlkj6612** [comment]
New code is in https://github.com/vvto33/yt-dlp/pull/1. What's the easiest and quickest way to merge my code into this PR?

**Comment by bashonly** [comment]
@pzhlkj6612 I applied your changes plus some additional refactoring. Please test my changes, thanks

**Review by pzhlkj6612 — COMMENTED** [review]
@bashonly, the current code does work!

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/tver.py [hunk 1]
```diff
@@ -21,80 +22,115 @@ class TVerIE(InfoExtractor):
...
-        self._PLATFORM_TOKEN = traverse_obj(create_response, ('result', 'platform_token'))
+
+    def _yield_episode_ids_for_series(self, series_id):
+        seasons_info = self._download_json(
+            f'https://service-api.tver.jp/api/v1/callSeriesSeasons/{series_id}',
+            series_id, 'Downloading seasons info', headers=self._HEADERS)
+        for season_id in traverse_obj(
+                seasons_info, ('result', 'contents', lambda _, v: v['type'] == 'season', 'content', 'id', {str})):
+            episodes_info = self._call_platform_api(
+                f'v1/callSeasonEpisodes/{season_id}', season_id, f'Downloading season {season_id} episodes info')
```
Comment (bashonly) [inline yt_dlp/extractor/tver.py:102]:
```suggestion
            episodes_info = self._call_platform_api(
                f'v1/callSeasonEpisodes/{season_id}', series_id, f'Downloading season {season_id} episodes info')
```

**Review by bashonly — APPROVED** [review]

**Review by Grub4K — APPROVED** [review]

**Pull request merged**